### PR TITLE
Add given name and phone

### LINKED ISSUE #191: parse CSR DirectoryName,  subject
I have created a CSR using pkijs the subject having many fields.
GN = Bob+SN = Test+telephoneNumber = 50655555555+emailAddress = test@eample.com+CN = Test CSR+OU = Dev+O = Example Com+L = Zaragosa+ST = Alajula+C = CR

let bcsr = Csr::from_pem_str(&csrStr).unwrap()
let csrName = bcsr.subject_name();

Here is the first part of the csrName
DirectoryName(Name(Asn1SequenceOf([Asn1SetOf([AttributeTypeAndValue { ty: ObjectIdentifierAsn1(ObjectIdentifier { root: JointIsoItuT, first_node: 5, child_nodes: [4, 6] }), value: CountryName(PrintableString(PrintableStringAsn1(RestrictedString { data: [67, 82], marker: PhantomData }))) }, AttributeTypeAndValue { ty: ObjectIdentifierAsn1(ObjectIdentifier { root: JointIsoItuT, first_node: 5, child_nodes: [4, 8] }), value: StateOrProvinceName(Utf8String("Alajula")) }, .....

So I have a Vec<AttributeTypeAndValue> with a ty: 2.5.4.6 and a value: CR  * Costa Rica not Carriage Return 
as the first entry in Vec

Is there a way to get all the fields in a Vec?

### PR DESCRIPTION
**Pull request by greenpdx**
I wanted to add a phone number and given name

Closes #191

### CODE REVIEW

**Comment by awakecoding** [comment]
@CBenoit

File: picky-asn1-x509/src/attribute_type_and_value.rs [hunk 1]
```diff
@@ -94,6 +96,18 @@ impl AttributeTypeAndValue {
...
+            ty: oids::at_given_name().into(),
+            value: AttributeTypeAndValueParameters::GivenName(name.into()),
+        }
+    }
+    pub fn new_phone<S: Into<DirectoryString>>(name: S) -> Self {
+        Self {
+            ty: oids::at_phone().into(),
+            value: AttributeTypeAndValueParameters::Phone(name.into()),
+        }
+    }
```
Comment (CBenoit) [inline picky-asn1-x509/src/attribute_type_and_value.rs:110]: Nitpick. Could you add a line break between function declarations?

**Review by CBenoit — APPROVED** [review]
Thank you! This looks good to me :100:
I’ll let CI run and merge tomorrow

File: picky-asn1-x509/src/attribute_type_and_value.rs [hunk 2]
```diff
@@ -94,6 +96,18 @@ impl AttributeTypeAndValue {
             value: AttributeTypeAndValueParameters::EmailAddress(name.into()),
         }
     }
+    pub fn new_given_name<S: Into<DirectoryString>>(name: S) -> Self {
```
Comment (CBenoit) [inline picky-asn1-x509/src/attribute_type_and_value.rs:99]:
```suggestion

    pub fn new_given_name<S: Into<DirectoryString>>(name: S) -> Self {
```

File: picky-asn1-x509/src/attribute_type_and_value.rs [hunk 3]
```diff
@@ -94,6 +96,19 @@ impl AttributeTypeAndValue {
...
         }
     }
+
+    pub fn new_given_name<S: Into<DirectoryString>>(name: S) -> Self {
+        Self {
+            ty: oids::at_given_name().into(),
+            value: AttributeTypeAndValueParameters::GivenName(name.into()),
+        }
+    }
+    pub fn new_phone<S: Into<DirectoryString>>(name: S) -> Self {
```
Comment (CBenoit) [inline picky-asn1-x509/src/attribute_type_and_value.rs:106]:
```suggestion

    pub fn new_phone<S: Into<DirectoryString>>(name: S) -> Self {
```

**Pull request merged**